### ISSUE TITLE
tf.while_loop seg faults upon setting parallel_iterations=0

### ISSUE THREAD

**Issue by cdjkim** [tensorflow/tensorflow#5257]
NOTE: Only file GitHub issues for bugs and feature requests.  All other topics will be closed.

For general support from the community, see [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow).
To make bugs and feature requests more easy to find and organize, we close issues that are deemed
out of scope for GitHub Issues and point people to StackOverflow.

For bugs or installation issues, please provide the following information.
The more information you provide, the more easily we will be able to offer
help and advice.
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?

Haven't found segmentation error related issue with tf.while_loop yet.
### Environment info

Operating System:

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
cuda-7.5
-rw-r--r-- 1 root root 189170 Jun 14 14:21 libcudadevrt.a
lrwxrwxrwx 1 root root     16 Jun 14 14:21 libcudart.so -> libcudart.so.7.5
lrwxrwxrwx 1 root root     19 Jun 14 14:21 libcudart.so.7.5 -> libcudart.so.7.5.18
-rwxr-xr-x 1 root root 311596 Jun 14 14:21 libcudart.so.7.5.18
-rw-r--r-- 1 root root 558020 Jun 14 14:21 libcudart_static.a

If installed from binary pip package, provide:
1. A link to the pip package you installed:
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`.

If installed from source, provide 
1. The commit hash (`git rev-parse HEAD`)
2. The output of `bazel version`
### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

``` python
import tensorflow as tf                                                                                                                                                                        
import numpy as np                                                                                                                                                                             

i = tf.constant(0)                                                                                                                                                                             
s = tf.constant(0)                                                                                                                                                                             
x = tf.random_uniform([], 0, 761-64, dtype=tf.int32)                                                                                                                                           
p = tf.Print(x, [x], message="This is random x: ")                                                                                                                                             
c = lambda i,s,x,p: tf.less(i, 50)                                                                                                                                                             
b = lambda i,s,x,p: (tf.add(i, 1), tf.add(s, x),tf.random_uniform([], 0,                                                                                                                       
    761-64, dtype=tf.int32),tf.Print(x, [x], message="This is random x: "))                                                                                                                    
r = tf.while_loop(c, b, [i,s,x,p], parallel_iterations=1)

sess = tf.Session()                                                                                                                                                                            
init = tf.initialize_all_variables().run                                                                                                                                                       
print sess.run(r)
```
### What other attempted solutions have you tried?

only ways which do not use the while loop.
### Logs or other output that would be helpful

(If logs are large, please upload as attachment or provide link).

error output:
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcudnn.so.4 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:108] successfully opened CUDA library libcurand.so locally
I tensorflow/core/common_runtime/gpu/gpu_init.cc:102] Found device 0 with properties:
name: GeForce GTX TITAN X
major: 5 minor: 2 memoryClockRate (GHz) 1.076
pciBusID 0000:04:00.0
Total memory: 11.92GiB
Free memory: 11.81GiB
I tensorflow/core/common_runtime/gpu/gpu_init.cc:126] DMA: 0
I tensorflow/core/common_runtime/gpu/gpu_init.cc:136] 0:   Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:806] Creating TensorFlow device (/gpu:0) -> (device: 0, name: GeForce GTX TITAN X, pci bus id: 0000:04:00.0)
Segmentation fault (core dumped)

**Comment by drpngx** [comment]
Are you running on Mac?

**Comment by cdjkim** [comment]
I'm running on Ubuntu 14.04.5 LTS

**Comment by drpngx** [comment]
Does it SEGV on CPU as well?

**Comment by darrengarvey** [comment]
@cdjkim - Might be a copy-paste error but it doesn't look like you're actually running the init op. Rather than:

    sess = tf.Session()
    init = tf.initialize_all_variables().run
    print sess.run(r)
try:

    sess = tf.Session()
    sess.run(tf.initialize_all_variables())
    print sess.run(r)

**Comment by drpngx** [comment]
Thanks @darrengarvey , nice find.

It should not crash and that's definitely a bug.

**Comment by yuanbyu** [comment]
`parallel_iterations` is the number of iterations allowed to run in parallel. So setting it to 1 means running the iterations sequentially. Setting it to 0 makes no sense. We will add a check and print a proper error message.
